zfs.block_mode or initial.zfs.block_mode broken?

Hi,

I've been trying to implement this for a couple of containers that will be hosting docker swarm services to try and eliminate the IO bottleneck on ZFS, however using this device block:

```
  device {
    name = "root"
    type = "disk"
    properties = {
      "path"           = "/"
      "pool"           = "local"
      #"size"           = "200GiB"
      "initial.zfs.block_mode" = true
    }
  }
```

Results in the following errors:
```
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to lxd_instance.int-dev-zvol["int-dev-zvol01"], provider "provider[\"registry.terraform.io/terraform-lxd/lxd\"]"
│ produced an unexpected new value: .device: actual set element cty.ObjectVal(map[string]cty.Value{"name":cty.StringVal("root"),
│ "properties":cty.MapVal(map[string]cty.Value{"initial.zfs.block_mode":cty.StringVal("true"), "path":cty.StringVal("/"),
│ "pool":cty.StringVal("local")}), "type":cty.StringVal("disk")}) does not correlate with any element in plan.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to lxd_instance.int-dev-zvol["int-dev-zvol01"], provider "provider[\"registry.terraform.io/terraform-lxd/lxd\"]"
│ produced an unexpected new value: .device: block set length changed from 1 to 2.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
```

This does appear to create the zvol block devices as desired, however:

```
rpool/lxd/containers/int-dev-swarm01                                                       946M  2.08T     1.74G  legacy
rpool/lxd/containers/int-dev-swarm02                                                       947M  2.08T     1.74G  legacy
rpool/lxd/containers/int-dev-worker01                                                      947M  2.08T     1.74G  legacy
rpool/lxd/containers/int-dev-zvol01                                                       1.03G  2.08T     1.03G  -
rpool/lxd/virtual-machines/int-dev-lab04.block                                            3.76G  2.08T     4.52G  -
```
(The 'legacy' ones are normal datasets, and lab04 has a manually created, attached and functional ext4 block device)

Am I omitting something basic? Essentially I just need the provider to create zvol block devices for specific container instances rather than zfs datasets when specified.

If I manually set the pool to use zfs.block_mode this works without issue, without the disk device being specified.

Versions:
lxd 5.19
lxd provider 2.3.0
terraform 1.5.3

Relevant TF_LOG=debug output:
```
2024-09-20T18:02:22.044+0100 [ERROR] provider.terraform-provider-lxd_v2.3.0: Response contains error diagnostic: diagnostic_detail="Failed instance creation: Failed creating instance from image: Could not locate a zvol for rpool/lxd/containers/int-dev-manager01" diagnostic_severity=ERROR diagnostic_summary="Failed to create instance "int-dev-manager01"" tf_proto_version=6.6 tf_rpc=ApplyResourceChange tf_req_id=7ac2f8a3-3dbc-0e05-edcb-c6b639747152 tf_resource_type=lxd_instance @caller=github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov6/internal/diag/diagnostics.go:58 @module=sdk.proto tf_provider_addr=registry.terraform.io/terraform-lxd/lxd timestamp=2024-09-20T18:02:22.044+0100
2024-09-20T18:02:22.046+0100 [DEBUG] State storage *remote.State declined to persist a state snapshot
2024-09-20T18:02:22.046+0100 [ERROR] vertex "lxd_instance.int-dev-manager[\"int-dev-manager01\"]" error: Failed to create instance "int-dev-manager01"
```

--- a/docs/resources/volume.md
+++ b/docs/resources/volume.md
@@ -30,7 +30,16 @@ resource "lxd_volume" "volume" {
   name = "myvol"
   pool = lxd_storage_pool.pool.name
 
+  # Set content_type to "filesystem" (default) to create a filesystem.
+  # Set content_type to "block" to create a block device without a filesystem.
+  # Note: Block volumes can only be attached to virtual machines.
+  content_type = "filesystem"
+
   config = {
+    # Some drivers support block-backed filesystem volumes.
+    # Note: This is a storage driver specific option.
+    "zfs.block_mode" = true
+
     size = "5GiB"
   }
 }

